include/zephyr/kernel.h

```diff
@@ -213,6 +213,38 @@ extern void k_thread_foreach_unlocked(
  */
 #define K_CALLBACK_STATE (BIT(4))
 
+#ifdef CONFIG_ARC
+/* ARC processor Bitmask definitions for threads user options */
+#define K_FP_IDX 1
```

Comment:
Let's move it from ARC options to K_FP_REGS definition itself. So we'll get:

```
#if defined(CONFIG_FPU_SHARING)
/**
 * @brief FPU registers are managed by context switch
 *
 * @details
 * This option indicates that the thread uses the CPU's floating point
 * registers. This instructs the kernel to take additional steps to save
 * and restore the contents of these registers when scheduling the thread.
 * No effect if @kconfig{CONFIG_FPU_SHARING} is not enabled.
 */
#define K_FP_IDX 1
#define K_FP_REGS (BIT(K_FP_IDX))
#endif
```

in `include/zephyr/kernel.h`